packages/dds/tree/src/test/simple-tree/simpleTree.bench.ts

```diff
@@ -80,6 +82,266 @@ describe("SimpleTree benchmarks", () => {
 				},
 			});
 		}
+
+		function testAccessToLeaf<RootSchema>(
+			title: string,
+			treeInitFunction: () => RootSchema,
+			treeReadingFunction: (tree: RootSchema) => number | undefined,
+			expectedValue: number | undefined,
+		) {
+			let tree: RootSchema;
+			let readNumber: number | undefined;
+			benchmark({
+				type: BenchmarkType.Measurement,
+				title,
+				before: () => {
+					tree = treeInitFunction();
+				},
+				benchmarkFn: () => {
+					readNumber = treeReadingFunction(tree);
+				},
+				after: () => {
+					assert.equal(readNumber, expectedValue);
+				},
+			});
+		}
+
+		describe("Access to leaves", () => {
+			describe("Optional object property", () => {
+				const factory = new SchemaFactory("test");
+				class MySchema extends factory.object("root", {
+					value: factory.optional(factory.number),
+					leafUnion: factory.optional([factory.number, factory.string]),
+					complexUnion: factory.optional([
+						factory.number,
+						factory.object("inner", {
+							value: factory.optional(factory.number),
+						}),
+					]),
+				}) {}
+
+				const testCases = [
+					{
+						title: `Read value from leaf`,
+						initUnhydrated: () => new MySchema({ value: 1 }),
+						initFlex: () => hydrate(MySchema, { value: 1 }),
+						read: (tree: MySchema) => tree.value,
+						expected: 1,
+					},
+					{
+						title: `Read value from union of two leaves`,
+						initUnhydrated: () => new MySchema({ leafUnion: 1 }),
+						initFlex: () => hydrate(MySchema, { leafUnion: 1 }),
```

Comment:
there is no need to provide `initFlex` functions for all of these. Sicne the schem is the same for all of them you can just do `() => hydrate(MySchema, initUnhydrated())` below (hydrate should be able to take in an actual node instance, or the constructor data)